### PR TITLE
Fixed ITHC issue

### PR DESCRIPTION
**Pull request by iceraluk**
Open redirect issue now fixed
Also fixed unsuccessful redirects on sign in the cases in which account is not confirmed or when it requires password reset

### CODE REVIEW

File: app/services/cognito/sign_in_user.rb [hunk 1]
```diff
@@ -77,5 +79,9 @@ def login_user_cognito(client_id, client_secret)
     def cookies_should_be_enabled
       errors.add(:base, :cookies_disabled) if @cookies_disabled
     end
+
+    def redirect_uri_known
+      errors.add(:base, :redirect_uri_mismatch) unless ENV['CALLBACK_URLS'].include?(@redirect_uri)
```
Comment (tim-s-ccs) [inline app/services/cognito/sign_in_user.rb:84]: What are the callback URLs?

Comment (iceraluk) [reply app/services/cognito/sign_in_user.rb:84]: The supplier Callback URL set up in Cognito. On Tail Spend we have 2, that is why this is an array

Comment (tim-s-ccs) [reply app/services/cognito/sign_in_user.rb:84]: What ITHC issue is this fixing out of curiosity?

Comment (iceraluk) [reply app/services/cognito/sign_in_user.rb:84]: https://crowncommercialservice.atlassian.net/browse/TS-133

For some reason this didn't come up in the PMP ITHC

Comment (tim-s-ccs) [reply app/services/cognito/sign_in_user.rb:84]: Thanks Iullia, I will do these updates in PMP too

**Review by tim-s-ccs — APPROVED** [review]

**Pull request merged**